src/content/en/2021/page-weight.md

```diff
@@ -1,21 +1,176 @@
+
+**Note**: If you need an online service that will optimize your images, there is no better source I've found than Google's [Squoosh](https://squoosh.app/) application, and Jake Archibald's [SVGOMG](https://jakearchibald.github.io/svgomg/) for optimizing SVG's.
+
+#### A word about the proliferation in the use of JavaScript
+JavaScript can be a wonderful tool to use for creating a dynamic website, but using it unchecked can create serious performance problems and a horrible experience for the user. There's been a proliferation in the use of complex JavaScript web frameworks and libraries over the past decades. Some of them can cause build sizes for a site to skyrocket leading to serious performance bottlenecks. Some so bad that a site can become unstable or even unusable. Blocking scripts, defined as a request of a script that must be transmitted, processed, and executed before the page can finish rendering enough page assets for users to interact with it. That can cause confusion, frustration, and abandonment by the user.
+
+Example: You navigate to a e-commerce site on your nifty smartphone on slow 4G internet connection, but the site is taking forever to load. It even throws a warning that the page is taking too long to load and asks if you want to continue waiting. Your smartphone or computer stalls and may even need to be restarted. Nine times out of ten a blocking JavaScript that is causing your smartphone to run out of processing resources or memory is to blame. The judicious, expert use of JavaScript can create great user experiences. **But remember this**: *JavaScript is executed on the client side. It's using the client computers resources to process and execute the script, and there is a finite amount of resources on every device*. Once again, not everyone is glued to the newest Google Pixel or Apple smartphone. The [JavaScript](https://almanac.httparchive.org/en/2021/javascript) chapter contains a wealth of information about this issue.
+
+### Dynamic
+Dynamically generated assets normally are produced by web programming languages like PHP, JavaScript, etc. Like other files, a requests is sent to locate, process, and retrieve the results of the scripts. The payload is returned to the user's browser for rendering. Generally, dynamic scripting in modern web sites execute extremely fast, and their results can also be cached for even faster use in future requests. But poorly scripted dynamic assets can have the opposite affect, increasing page weight. So, for best results, use best practices and and apply optimizations.
+
+### Connected
```

Comment:
Not sure I understand this one either. Are you saying "connected" is a type of asset? Are you talking more about 3rd party assets?